### PR TITLE
Adding pod env command to print environment

### PR DESCRIPTION
**Pull request by hemal**
Closes https://github.com/CocoaPods/CocoaPods/issues/3660.

### CODE REVIEW

**Comment by hemal** [comment]
For this issue: https://github.com/CocoaPods/CocoaPods/issues/3660

**Comment by hemal** [comment]
Preview of output: 

screen shot 2015-11-11 at 11 10 40 am

File: lib/cocoapods/command/env.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+require 'cocoapods/user_interface/error_report'
+
+module Pod
+  class Command
+    class Env < Command
+      def run
+        puts "
+#{'――― MARKDOWN TEMPLATE ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――'.reversed}
```
Comment (segiddins) [inline lib/cocoapods/command/env.rb]: The template itself shouldn't be duplicated, and the header and footer shouldn't be printed by the command, only by the error report.

File: lib/cocoapods/command/env.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      def markdown_podfile
+        UI::ErrorReport.markdown_podfile
+      end
+
+      def plugins_string
+        UI::ErrorReport.plugins_string
+      end
+
+      def executable_path
+        `which pod`
```
Comment (segiddins) [inline lib/cocoapods/command/env.rb]: this is wrong, use what the error report does -- i believe its `$0`

**Comment by segiddins** [comment]
Much better! Do you think we could change the multi-line strings to use heredocs (`<<-EOS`), change the method names to not say `print` iff they return a string, and add some specs for the new command? Then it'll be good to go!

**Comment by segiddins** [comment]
Sweet, let's just rebase this and add a CHANGELOG entry? Thanks :)

File: spec/cocoapods-integration-specs [hunk 3]
```diff
@@ -1 +1 @@
-Subproject commit 10f25aa8855b5230f9068a8f6db3a8ebff67efd6
+Subproject commit ad385c014ac0c5f5b28fd22aa7c136554b9b6bf0
```
Comment (segiddins) [inline spec/cocoapods-integration-specs]: why have these been changed?

File: spec/unit/command_spec.rb [hunk 4]
```diff
@@ -16,6 +16,7 @@ module Pod
       Command.parse(%w(spec create  )).should.be.instance_of Command::Spec::Create
       Command.parse(%w(spec lint    )).should.be.instance_of Command::Spec::Lint
       Command.parse(%w(init         )).should.be.instance_of Command::Init
+      Command.parse(%w(env         )).should.be.instance_of Command::Env
```
Comment (segiddins) [inline spec/unit/command_spec.rb]: alignment

File: spec/unit/env_spec.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
+require File.expand_path('../../../spec_helper', __FILE__)
```
Comment (segiddins) [inline spec/unit/env_spec.rb]: belongs in the `command` directory

File: spec/unit/env_spec.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+module Pod
+  describe Env do
```
Comment (segiddins) [inline spec/unit/env_spec.rb]: `Command::Env`

File: lib/cocoapods/command/env.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
+require 'cocoapods/user_interface/error_report'
+
+module Pod
+  class Command
+    class Env < Command
```
Comment (segiddins) [inline lib/cocoapods/command/env.rb]: should override `self.options` and reject `--silent`

File: lib/cocoapods/command/env.rb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+      def run
+        puts report
```
Comment (segiddins) [inline lib/cocoapods/command/env.rb]: `UI.puts`

File: spec/unit/env_spec.rb [hunk 9]
```diff
@@ -0,0 +1,54 @@
+require File.expand_path('../../spec_helper', __FILE__)
+
+module Pod
+  describe Command::Env do
+
+    describe 'In general' do
+      before do
+        @report = Env
```
Comment (segiddins) [inline spec/unit/env_spec.rb]: `Command::Env`

File: spec/unit/env_spec.rb [hunk 10]
```diff
@@ -0,0 +1,54 @@
+require File.expand_path('../../spec_helper', __FILE__)
+
+module Pod
```
Comment (segiddins) [inline spec/unit/env_spec.rb]: This file still needs to be moved to the correct directory

**Comment by segiddins** [comment]
:shipit: thanks so much, @hemal !

**Comment by orta** [comment]
@hemal :tada:

**Comment by hemal** [comment]
Thanks @segiddins and @orta! This was a ton of fun, I'm looking forward to doing it again :)

**Pull request merged**